Model/Adminhtml/Config/TriggerIsAllowed.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Alma\MonthlyPayments\Model\Adminhtml\Config;
+
+use Alma\MonthlyPayments\Helpers\PaymentPlansHelper;
+use Magento\Config\Model\Config\Source\Yesno;
+use Alma\MonthlyPayments\Helpers\Logger;
+
+class TriggerIsAllowed extends Yesno
+{
+    const ALLOWED_LABEL = 'Yes';
+    const NOT_ALLOWED_LABEL = 'No';
+    /**
+     * @var Logger
+     */
+    private $logger;
+    /**
+     * @var PaymentPlansHelper
+     */
+    private $paymentPlansHelper;
+
+    public function __construct(
+        Logger $logger,
+        PaymentPlansHelper $paymentPlansHelper
+    )
+    {
+        $this->logger = $logger;
+        $this->paymentPlansHelper = $paymentPlansHelper;
+    }
+
+    protected function _getElementHtml(AbstractElement $element)
+    {
+        $element->setDisabled('disabled');
+        return $element->getElementHtml();
+    }
+
+    /**
+     * Options getter
+     *
+     * @return array
+     */
+    public function toOptionArray()
+    {
+        $isAllowedValue = 0;
+        $isAllowedLabel = self::NOT_ALLOWED_LABEL;
+        if($this->paymentPlansHelper->paymentTriggerIsAllowed()){
+            $isAllowedValue = 1;
+            $isAllowedLabel = self::ALLOWED_LABEL;
+        }
+        return [['value' => $isAllowedValue, 'label' => __($isAllowedLabel)]];
```

Comment:
warning: translate a variable don't will add translation key in our translations files !

Reply:
it's "yes" or "no"  label they are already translated in default